board: add reel_board and reel_board_v2

jremmert-phytec-iot opened the pull request:
This PR adds support for reel_board and reel_board_v2. The main platform for the reel_board is Zephyr. More information can be found in the [documentation page](https://docs.zephyrproject.org/latest/boards/arm/reel_board/doc/index.html) of the reel board within the Zephyr project.

I tested the boards here locally and building, debugging and serial monitor works for me. Is there a way to make a baudrate of 115200 the default for this board? Zephyr uses 115200 for most of the examples.

reel_board and reel_board_v2 differ in the passive display being used. I created two different boards. Not sure if there is a more elegant way to go?

valeros commented on boards/reel_board.json:
```diff
@@ -0,0 +1,44 @@
+{
+  "build": {
+    "core": "nRF5",
```
This field is mostly used with Arduino framework. nRF5 core doesn't support this board, so it seems redundant to me.

valeros commented on boards/reel_board.json:
```diff
@@ -0,0 +1,44 @@
+    "core": "nRF5",
+    "cpu": "cortex-m4",
+    "f_cpu": "64000000L",
+    "mcu": "nrf52832",
```
Looks like you forget to change the MCU name

valeros commented on boards/reel_board.json:
```diff
@@ -0,0 +1,44 @@
+    "f_cpu": "64000000L",
+    "mcu": "nrf52832",
+ 	"hwids": [
+		[
+        "0x0D28",
+        "0x0204"
+      	]
```
Please fix formatting.

valeros commented on boards/reel_board.json:
```diff
@@ -0,0 +1,44 @@
+        "0x0204"
+      	]
+    ],
+   "variant": "reel_board"
```
Please move the `variant` field to the `zephyr` section as it done in [this](https://github.com/platformio/platform-nordicnrf52/blob/develop/boards/laird_bl652_dvk.json#L7) manifest.

valeros commented:
Hi @jremmert-phytec-iot ! Thanks for the PR. Please see my comments above, they apply to both boards.

jremmert-phytec-iot commented:
> Hi @jremmert-phytec-iot ! Thanks for the PR. Please see my comments above, they apply to both boards.

Thanks for review @valeros! I addressed all your comments and did a force push.

valeros commented:
Many thanks!
